tooling/nargo_cli/build.rs

```diff
@@ -89,9 +111,88 @@ const TESTS_WITH_EXPECTED_WARNINGS: [&str; 4] = [
     "comptime_enums",
 ];
 
+/// These tests are ignored because making them work involves a more complex test code that
+/// might not be worth it.
+/// Others are ignored because of existing bugs in `nargo expand`.
+/// As the bugs are fixed these tests should be removed from this list.
+const IGNORED_NARGO_EXPAND_EXECUTION_TESTS: [&str; 6] = [
+    // There's nothing special about this program but making it work with a custom entry would involve
+    // having to parse the Nargo.toml file, etc., which is not worth it
+    "custom_entry",
+    // There's no "src/main.nr" here so it's trickier to make this work
+    "diamond_deps_0",
+    // There's no "src/main.nr" here so it's trickier to make this work
+    "overlapping_dep_and_mod",
+    // bug
+    "poseidonsponge_x5_254",
+    // There's no "src/main.nr" here so it's trickier to make this work
+    "workspace",
+    // There's no "src/main.nr" here so it's trickier to make this work
+    "workspace_default_member",
+];
+
 /// Tests for which we don't check that stdout matches the expected output.
 const TESTS_WITHOUT_STDOUT_CHECK: [&str; 0] = [];
 
+/// These tests are ignored because of existing bugs in `nargo expand`.
+/// As the bugs are fixed these tests should be removed from this list.
+/// (some are ignored on purpose for the same reason as `IGNORED_NARGO_EXPAND_EXECUTION_TESTS`)
+const IGNORED_NARGO_EXPAND_COMPILE_SUCCESS_EMPTY_TESTS: [&str; 15] = [
+    // There's no "src/main.nr" here so it's trickier to make this work
+    "overlapping_dep_and_mod",
+    // bug
+    "reexports",
+    // bug
+    "regression_4436",
+    // bug
+    "regression_7038",
+    // bug
+    "regression_7038_2",
+    // bug
+    "regression_7038_3",
+    // bug
+    "regression_7038_4",
+    // bug
+    "serialize",
+    // bug
+    "trait_allowed_item_name_matches",
+    // bug
+    "trait_default_implementation",
+    // bug
+    "trait_function_calls",
+    // bug
+    "trait_method_mut_self",
+    // bug
+    "trait_override_implementation",
+    // bug
+    "trait_static_methods",
+    // There's no "src/main.nr" here so it's trickier to make this work
+    "workspace_reexport_bug",
+];
+
+/// These tests are ignored because of existing bugs in `nargo expand`.
+/// As the bugs are fixed these tests should be removed from this list.
+const IGNORED_NARGO_EXPAND_COMPILE_SUCCESS_NO_BUG_TESTS: [&str; 15] = [
```

Comment:
follow up issue(s) for the ignored tests?

Reply:
Done! https://github.com/noir-lang/noir/issues/8281